libkbfs/disk_block_cache_wrapped.go

```diff
@@ -112,23 +111,43 @@ func (cache *diskBlockCacheWrapped) IsSyncCacheEnabled() bool {
 }
 
 // Get implements the DiskBlockCache interface for diskBlockCacheWrapped.
-func (cache *diskBlockCacheWrapped) Get(ctx context.Context, tlfID tlf.ID,
-	blockID kbfsblock.ID) (
+func (cache *diskBlockCacheWrapped) Get(
+	ctx context.Context, tlfID tlf.ID, blockID kbfsblock.ID,
+	preferredCacheType DiskBlockCacheType) (
```

Comment:
This is not an atomic operation. Stuff can happen between the Get-Get-Put-UpdateMetadata-Delete sequence. I can't think of an actual bug that would happen with this yet, but thought I'd bring it up. Also, if you decide to not make it atomic, perhaps worth mentioning in comment in `interfaces.go`?

Reply:
Except for the final delete, it's protected with `cache.mtx.RLock()`.  Did you miss that, or were you worried about two Gets happening at the same time?

Reply:
The worst I can think of is undoing a Delete that happens in the middle of the Get operations, so a block that should be deleted ends up moved to a sync cache instead.  That seems pretty unlikely since explicit deletes happen only after archiving, and the block shouldn't be getting accessed after it's archived.

So I think it's probably safe as is, but would like @jzila to weigh in as well.

Reply:
@jzila Yes agreed. But what do you think about the race possibilities here?  Do you think it's worth having locks to protect against the unlikely delete race I mention above?  Or can you think of any other races to be worried about?

Reply:
When this was first built, I think we didn't care about that particular race--the worst case is that sometimes we have a block in both caches, or we have a block in a cache when it shouldn't be any more. And as you say, an archived block shouldn't be accessed any more so this is extremely unlikely.